api/src/main/java/org/pmiops/workbench/cohortbuilder/querybuilder/FactoryKey.java

```diff
@@ -0,0 +1,39 @@
+package org.pmiops.workbench.cohortbuilder.querybuilder;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * This enum maps specific implementations of {@link AbstractQueryBuilder} to specific
+ * criteria types/searches. This enum is built specifically for use of the
+ * {@link org.pmiops.workbench.cohortbuilder.QueryBuilderFactory} to get the correct
+ * factory implementation.
+ */
+public enum FactoryKey {
+    CRITERIA("criteria-type",  Arrays.asList("ICD9-TREE", "ICD10-TREE", "CPT-TREE")),
```

Comment:
Can you add some comments describing the types of queries performed for each of these?

If some of these are for fetching criteria and others are for searching for participants(?), it's a little confusing to have them lumped together here... we should know in the code when we're dealing with one or the other.

Reply:
Yes, i have both criteria and searching mixed. I will clean this up.